test/docfx.Test/e2e/E2ETest.cs

```diff
@@ -267,6 +272,12 @@ private static TheoryData<string> FindTestSpecs()
             }
 
             var docset = Path.Combine(inputFolder, spec.Cwd ?? string.Empty);
+
+            if (GitUtility.IsRepo(docset) && !(spec.Inputs?.Any() ?? false))
+            {
+                Process.Start(new ProcessStartInfo("git", "clean -fdx") { WorkingDirectory = docset }).WaitForExit();
```

Comment:
This is likely going to make test slower, how about removing output files starting with `~/` instead

Reply:
not just remove, but make the change revert

Reply:
then only run this command if the output has any files with `~/`. Plus, `spec.Inputs` should never be `null`.

Reply:
👍